### PR TITLE
Pre commit hook, drop stairs config for .env and bin/setup

### PR DESCRIPTION
**Pull request by ashwinv11**
Updated the README with info on how to skip this, and how to install the hooks when another user clones the project

### CODE REVIEW

**Comment by patbenatar** [comment]
@ashwinv11 On further thought, I don't think this is going to work for our use case. Nothing in the `.git/` directory gets version controlled or moved around with the repo — i.e. after running suspenders you'll have a git commit hook in your local repo but when you push to GitHub, that hook won't be pushed, so the next person to clone won't get the hook. This is starting to justify the use of a tool to manage git hooks in the repo... see https://stackoverflow.com/questions/427207/can-git-hook-scripts-be-managed-along-with-the-repository

File: lib/suspenders/app_builder.rb [hunk 1]
```diff
@@ -432,16 +432,25 @@ def gitignore_files
...
+      bundle_command 'exec overcommit --sign pre-commit'
     end
 
     def setup_deployment_environment_branches
       run 'git branch staging'
       run 'git branch production'
     end
 
+    def run_rubocop_autofixes
+      run 'rm -rf test/'
```
Comment (ashwinv11) [inline lib/suspenders/app_builder.rb]: @patbenatar do we want to keep this folder?

**Review by ashwinv11 — COMMENTED** [review]
Just updated this to use the [overcommit gem](https://github.com/brigade/overcommit), sorry about the many commits, things were intermittently failing

File: README.md [hunk 2]
```diff
@@ -157,10 +157,12 @@ By default, suspenders will:
 * Create `staging` and `production` branches for deployment
 * Make an initial commit
 
+By default, pre-commit hooks are setup to run `bin/rails`. To override this while committing pass the `--no-verify` option to your commit.
```
Comment (patbenatar) [inline README.md:158]: I think you mean `bin/rake` ?

File: templates/.overcommit.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+CommitMsg:
+  ALL:
+    enabled: false
+
+PreCommit:
+  Rake:
+    enabled: true
+    command: ['bundle', 'exec', 'bin/rails']
```
Comment (patbenatar) [inline templates/.overcommit.yml]: `bin/rake` ? Does `bin/rails` just alias to `bin/rake` in Rails5?

File: lib/suspenders/app_builder.rb [hunk 4]
```diff
@@ -432,16 +432,25 @@ def gitignore_files
 
     def init_git
       run 'git init'
+      template '.overcommit.yml', '.overcommit.yml'
+      bundle_command 'exec overcommit --install'
+      bundle_command 'exec overcommit --sign'
+      bundle_command 'exec overcommit --sign pre-commit'
```
Comment (patbenatar) [inline lib/suspenders/app_builder.rb:451]: With overcommit, does every developer who clones the repo need to run anything in order to get it to bootstrap the  hooks in their local repo? Or does it magically do that somehow? (If so: any idea how?)

File: README.md (hunk 2 above)
Comment (patbenatar) [reply README.md:158]: Can you also add the `no-verify` note to the README template (the one that gets copied into the newly created Rails app)

File: lib/suspenders/app_builder.rb (hunk 4 above)
Comment (ashwinv11) [reply lib/suspenders/app_builder.rb:451]: Nope! Now that the `overcommit` gem is added and these steps run, all the user needs to do is commit normally and they get all the features of the pre-commit hooks. These steps are what installs the hooks in the `.git` folder

File: templates/.overcommit.yml (hunk 3 above)
Comment (ashwinv11) [reply templates/.overcommit.yml]: Yup! More details here: https://github.com/rails/rails/issues/18878

**Comment by ashwinv11** [comment]
@patbenatar this is ready for review

**Comment by patbenatar** [comment]
:clap: :shipit: !

**Comment by patbenatar** [comment]
:+1: :shipit: @ashwinv11

**Pull request merged**